server/worker/api.go

```diff
@@ -0,0 +1,95 @@
+package worker
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+type Client struct {
+	AnalyticsUrl string
+	GrafanaUrl   string
+	Token        string
+	Logger       log.Logger
+	Filter       string
+}
+
+const contentTypeJson = "application/json"
+
+func (api *Client) Get(endpoint string) ([]byte, error) {
+	req, err := api.prepareRequest("GET", endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
```

Comment:
add more context to error message. Endpoint and maybe method could be useful to understand from the logs